rfd/0067-database-access-aws-redis

```diff
@@ -0,0 +1,276 @@
+
+### In-transit encryption (TLS)
+
+For security purposes, Teleport should ONLY
+support databases with in-transit encryption (TLS) enabled.
+
+[Enabling TLS on ElastiCache
+Redis](https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/in-transit-encryption.html#in-transit-encryption-enable)
+can be a complicated process, and all client applications must be updated at
+the same time. Therefore, Teleport should NOT automatically enable TLS for
+existing clusters. Redis servers without TLS are NOT supported, and users are
+recommended to enable TLS separately to enable Teleport database access.
+
+### Auth
+
+Redis can be configured with one of the following authentication methods, and
+only one at a time.
+
+#### Auth - Redis with no auth
+
+Redis can be configured without any auth method.
+
+"default" user is always used in this configuration, and no `auth` command is
+required upon a successful connection.
+
+#### Auth - Redis with Redis Auth
+
+[Redis Auth](https://redis.io/commands/auth/)
+uses a single token/password that must be shared for all client applications.
+
+"default" user is always used in this configuration, and users have to manually
+input `auth` command upon a successful connection. <sup>1<sup>
+
+#### Auth - Redis with ACL
+
+[Redis ACL](https://redis.io/docs/manual/security/acl/) (also known as RBAC for
+ElastiCache Redis) is introduced in Redis 6. Both ElastiCache and MemoryDB
+provide APIs to manage database users and user groups for ACL.
+
+As with all other databases, the database users should be created by "database
+admins" (by users, not Teleport) to assign desired permissions. 
+
+To provide better security than using static passwords, users can tag
+ElastiCache and MemoryDB users with label **`teleport.dev/managed: true`** for
+Teleport to manage the passwords for those database users.
+
+Database service discovers database users by listing users from the ElastiCache
+user group or MemoryDB ACL attached to the Redis database. For database users
+discovered with label **`teleport.dev/managed: true`**, passwords are randomly
+generated, periodically rotated (e.g. every 20 minutes plus jitter), and securely
+stored in the secret store (more on the secret store in a later section).
+
+The creation time of the current version of the password should always be
+checked before applying a password rotation. If the current version is created
+fairly recently (e.g. less than 15 minutes), it is most likely that the
+password has been updated by another service in HA mode, so there is no action
+required. If the current version is indeed "expired", the database service will
+perform a rotation. In cases where more than one database service tries to
+rotate the same user at the same time, the secret store implementation
+guarantees only the first caller succeeds.
+
+It has been found that it may take a few minutes for AWS to propagate user
+password changes to the Redis servers. To work around this, two versions of the
```

Comment:
> two versions of the passwords are set to be effective AT THE SAME TIME for the database user

Is this something that Elasticache/MemDB support out of the box?

Reply:
yes